### PR TITLE
Allow sending to implicit accounts (#931)

### PR DESCRIPTION
**Pull request by marcinbodnar**

### CODE REVIEW

**Review by kcole16 — APPROVED** [review]

File: src/components/accounts/AccountFormAccountId.js [hunk 1]
```diff
@@ -133,16 +133,21 @@ class AccountFormAccountId extends Component {
 
     handleCheckAvailability = (accountId, type) => (
         accountId
-            && !(
```
Comment (vgrichina) [inline src/components/accounts/AccountFormAccountId.js:136]: nit: I find it pretty hard to read very nested conditional expressions.

Should we convert stuff like this to use multiple statements (e.g. early return pattern)? Something like:

```
if (!accountId) {
   return false;
}
if (this.isImplictAccount()) {
   return type != 'create';
}
if (type == 'create') {
 ....   
}
```

File: src/components/accounts/AccountFormAccountId.js [hunk 2]
```diff
@@ -133,16 +133,21 @@ class AccountFormAccountId extends Component {
...
-                type === 'create' 
-                && (!this.handleAccountIdLengthState(accountId) 
-                && !this.checkAccountIdLength(accountId))
+            && this.isImplicitAccount()
```
Comment (vgrichina) [inline src/components/accounts/AccountFormAccountId.js:136]: feels very weird that method takes `accountId` and then we only check `this.state.accountId` here

File: src/components/accounts/AccountFormAccountId.js [hunk 3]
```diff
@@ -163,15 +168,24 @@ class AccountFormAccountId extends Component {
...
+    get implicitAccountRequestStatus() {
+        return {
+            success: false,
+            messageCode: 'account.available.implicitAccount'
+        }
+    }
+
     get requestStatusWithFormValidation() {
         return this.state.accountId
-            ? this.props.formLoader
```
Comment (vgrichina) [inline src/components/accounts/AccountFormAccountId.js:168]: see above about early return pattern

File: src/components/accounts/AccountFormAccountId.js [hunk 4]
```diff
@@ -163,15 +168,24 @@ class AccountFormAccountId extends Component {
         }
     }
 
+    get implicitAccountRequestStatus() {
+        return {
+            success: false,
```
Comment (vgrichina) [inline src/components/accounts/AccountFormAccountId.js:173]: I don't think we should represent implicit account warning as failure

**Comment by vgrichina** [comment]
Screen Shot 2020-10-06 at 5 44 56 PM

@corwinharrell feel like we need to adjust this dialog for sending money, doesn't seem to scale well for implicit accounts

**Comment by vgrichina** [comment]
@marcinbodnar can you fix the
"Go Bac
k" and "Transfe
r" right away?

File: src/components/accounts/AccountFormAccountId.js (hunk 1 above)
Comment (marcinbodnar) [reply src/components/accounts/AccountFormAccountId.js:136]: Sure thing

**Comment by marcinbodnar** [comment]
@vgrichina 
> @marcinbodnar can you fix the

I'm not seeing this but, also looks very weird like the text doesn't fit inside the button, but there is plenty of space. But anyway, I made small changes it should look fine now.

**Comment by marcinbodnar** [comment]
@vgrichina @corwinharrell 
> @corwinharrell feel like we need to adjust this dialog for sending money, doesn't seem to scale well for implicit accounts

I removed the second accountId (small gray font) as it seems unnecessary at all.

**Review by kcole16 — APPROVED** [review]

**Pull request merged**